> Hi @SD-13 I left a comment. Also, does the doctest pass now (as you asked a question in another thread)
> 
> The checks on this PR page doesn't run doctest. So it's important for the contributors to run it pray please.

That totally makes sense. I am still getting the error and I am giving the whole error log here. please help me to debug this. Thanks,


============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.3, pluggy-1.0.0 -- /home/pirate/Downloads/huggingFace/transformers/transformers/bin/python
cachedir: .pytest_cache
rootdir: /home/pirate/Downloads/huggingFace/transformers, configfile: setup.cfg
collected 1 item                                                               

src/transformers/models/vision_text_dual_encoder/configuration_vision_text_dual_encoder.py::transformers.models.vision_text_dual_encoder.configuration_vision_text_dual_encoder.VisionTextDualEncoderConfig FAILED

=================================== FAILURES ===================================
_ [doctest] transformers.models.vision_text_dual_encoder.configuration_vision_text_dual_encoder.VisionTextDualEncoderConfig _
062 
063     >>> # Accessing the model configuration
064     >>> config_vision = model.config.vision_config
065     >>> config_text = model.config.text_config
066 
067     >>> # Saving the model, including its configuration
068     >>> model.save_pretrained("my-model")
069 
070     >>> # loading model and config from pretrained folder
071     >>> vision_text_config = VisionTextDualEncoderConfig.from_pretrained("vit-bert")
UNEXPECTED EXCEPTION: OSError("vit-bert is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models'\nIf this is a private repository, make sure to pass a token having permission to this repo with `use_auth_token` or log in with `huggingface-cli login` and pass `use_auth_token=True`.")
Traceback (most recent call last):
  File "/home/pirate/Downloads/huggingFace/transformers/transformers/lib/python3.10/site-packages/huggingface_hub/utils/_errors.py", line 213, in hf_raise_for_status
    response.raise_for_status()
  File "/home/pirate/Downloads/huggingFace/transformers/transformers/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://huggingface.co/vit-bert/resolve/main/config.json
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/home/pirate/Downloads/huggingFace/transformers/src/transformers/utils/hub.py", line 409, in cached_file
    resolved_file = hf_hub_download(
  File "/home/pirate/Downloads/huggingFace/transformers/transformers/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 1053, in hf_hub_download
    metadata = get_hf_file_metadata(
  File "/home/pirate/Downloads/huggingFace/transformers/transformers/lib/python3.10/site-packages/huggingface_hub/file_download.py", line 1359, in get_hf_file_metadata
    hf_raise_for_status(r)
  File "/home/pirate/Downloads/huggingFace/transformers/transformers/lib/python3.10/site-packages/huggingface_hub/utils/_errors.py", line 242, in hf_raise_for_status
    raise RepositoryNotFoundError(message, response) from e
huggingface_hub.utils._errors.RepositoryNotFoundError: 401 Client Error. (Request ID: WiAMp3MzQkYIQuEIq-5Wj)

Repository Not Found for url: https://huggingface.co/vit-bert/resolve/main/config.json.
Please make sure you specified the correct `repo_id` and `repo_type`.
If the repo is private, make sure you are authenticated.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.10/doctest.py", line 1350, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest transformers.models.vision_text_dual_encoder.configuration_vision_text_dual_encoder.VisionTextDualEncoderConfig[8]>", line 1, in <module>
  File "/home/pirate/Downloads/huggingFace/transformers/src/transformers/configuration_utils.py", line 531, in from_pretrained
    config_dict, kwargs = cls.get_config_dict(pretrained_model_name_or_path, **kwargs)
  File "/home/pirate/Downloads/huggingFace/transformers/src/transformers/configuration_utils.py", line 558, in get_config_dict
    config_dict, kwargs = cls._get_config_dict(pretrained_model_name_or_path, **kwargs)
  File "/home/pirate/Downloads/huggingFace/transformers/src/transformers/configuration_utils.py", line 613, in _get_config_dict
    resolved_config_file = cached_file(
  File "/home/pirate/Downloads/huggingFace/transformers/src/transformers/utils/hub.py", line 424, in cached_file
    raise EnvironmentError(
OSError: vit-bert is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models'
If this is a private repository, make sure to pass a token having permission to this repo with `use_auth_token` or log in with `huggingface-cli login` and pass `use_auth_token=True`.
/home/pirate/Downloads/huggingFace/transformers/src/transformers/models/vision_text_dual_encoder/configuration_vision_text_dual_encoder.py:71: UnexpectedException
063     >>> # Accessing the model configuration
064     >>> config_vision = model.config.vision_config
065     >>> config_text = model.config.text_config
066 
067     >>> # Saving the model, including its configuration
068     >>> model.save_pretrained("my-model")
069 
070     >>> # loading model and config from pretrained folder
071     >>> vision_text_config = VisionTextDualEncoderConfig.from_pretrained("vit-bert")
072     >>> model = VisionTextDualEncoderModel.from_pretrained("vit-bert", config=vision_text_config)
UNEXPECTED EXCEPTION: NameError("name 'vision_text_config' is not defined")
Traceback (most recent call last):
  File "/usr/lib/python3.10/doctest.py", line 1350, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest transformers.models.vision_text_dual_encoder.configuration_vision_text_dual_encoder.VisionTextDualEncoderConfig[9]>", line 1, in <module>
NameError: name 'vision_text_config' is not defined
/home/pirate/Downloads/huggingFace/transformers/src/transformers/models/vision_text_dual_encoder/configuration_vision_text_dual_encoder.py:72: UnexpectedException
=========================== short test summary info ============================
FAILED src/transformers/models/vision_text_dual_encoder/configuration_vision_text_dual_encoder.py::transformers.models.vision_text_dual_encoder.configuration_vision_text_dual_encoder.VisionTextDualEncoderConfig
============================== 1 failed in 5.48s ===============================

--- a/src/transformers/models/vision_text_dual_encoder/configuration_vision_text_dual_encoder.py
+++ b/src/transformers/models/vision_text_dual_encoder/configuration_vision_text_dual_encoder.py
@@ -49,7 +49,7 @@ class VisionTextDualEncoderConfig(PretrainedConfig):
     Examples:
 
     ```python
-    >>> from transformers import VisionTextDualEncoderModel, VisionTextDualEncoderConfig, BertConfig, ViTConfig
+    >>> from transformers import ViTConfig, BertConfig, VisionTextDualEncoderConfig, VisionTextDualEncoderModel
 
     >>> # Initializing a BERT and ViT configuration
     >>> config_vision = ViTConfig()
@@ -65,7 +65,7 @@ class VisionTextDualEncoderConfig(PretrainedConfig):
     >>> config_text = model.config.text_config
 
     >>> # Saving the model, including its configuration
-    >>> model.save_pretrained("my-model")
+    >>> model.save_pretrained("vit-bert")
 
     >>> # loading model and config from pretrained folder
     >>> vision_text_config = VisionTextDualEncoderConfig.from_pretrained("vit-bert")

